06-building-a-program-start-to-finish/writing-tests/output/invertedIndex-spec.js

```diff
@@ -0,0 +1,205 @@
+
+  describe('InvertedIndex class, check all methods', () => {
+    it('should check that the class has a createIndex method', () => {
+      expect(typeof invertedIndex.createIndex).toBe('function');
+    });
+
+    it('should check that the class has a validateFile method', () => {
+      expect(typeof InvertedIndex.validateFile).toBe('function');
+    });
+
+    it('should check that the class has a tokenize method', () => {
+      expect(typeof InvertedIndex.tokenize).toBe('function');
+    });
+
+    it('should check that the class has a getIndex method', () => {
+      expect(typeof invertedIndex.getIndex).toBe('function');
+    });
+
+    it('should check that the class has a searchIndex method', () => {
+      expect(typeof invertedIndex.searchIndex).toBe('function');
+    });
+
+    it('should check that the class has a setIndex method', () => {
+      expect(typeof invertedIndex.setIndex).toBe('function');
+    });
+  });
+
+  describe('validateFile should check files', () => {
+    it('should check that the contents of the file to be uploaded is valid',
+    () => {
+      expect(InvertedIndex.validateFile(book)).toBeTruthy();
+    });
+
+    it('should return false for empty json files', () => {
+      expect(InvertedIndex.validateFile(empty)).toBeFalsy();
+    });
+
+    it('should return true if file has property "title" and "text" ', () => {
+      expect(InvertedIndex.validateFile(book)).toBeTruthy();
+    });
+
+    it('should return false if file does not have property "title" and "text"',
+     () => {
+       expect(InvertedIndex.validateFile(invalidFile)).toBeFalsy();
+     });
+
+    it('should return false if file is not an array of JSON object',
+     () => {
+       expect(InvertedIndex.validateFile(invalid)).toBeFalsy();
+     });
+
+    it('should return false if file is an empty array',
+     () => {
+       expect(InvertedIndex.validateFile(emptyFile)).toBeFalsy();
+     });
+
+    it('should return false if file is a JSON file but not' +
+    'an array of an array',
+     () => {
+       expect(InvertedIndex.validateFile(notValid)).toBeFalsy();
+     });
+  });
+
+  describe('Create Index', () => {
+    it('should return a msg if index is not created', () => {
+      const msg = 'Index not created';
+      expect(invertedIndex.createIndex(notValid)).toEqual(msg);
+    });
+  });
+
+  describe('Tokenize words', () => {
+    it('should check that tokens are split and in sorted order', () => {
+      let words = 'Hello Dear how are YOU';
+      const expectedTokens = ['are', 'dear', 'hello', 'how', 'you'];
+      words = InvertedIndex.tokenize(words);
+      expect(expectedTokens).toEqual(words);
+    });
+  });
+
+  describe('Generate Index', () => {
+    it('should verify that index has been created', () => {
+      expect(Object.keys(invertedIndex.getIndex('book.json')).length)
+      .toBeGreaterThan(0);
+    });
+
+    it('should check that index maps the string to the correct objects in json'
```

Comment:
We don't advice breaking strings with `+`. Please consult the style guides.

Also, this suite is confusing. You don't have a method that does "generation". This suite seems more like one meant to test the `getIndex` method.

Please consult @andela-angene for clarity on this,